nmpolicy/internal/resolver/replace.go

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 NMPolicy Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *	  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package resolver
+
+import (
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/ast"
+)
+
+func replace(inputState map[string]interface{}, path ast.VariadicOperator, expectedNode ast.Node) (map[string]interface{}, error) {
+	replaced, err := applyFuncOnPath(inputState, path, expectedNode, replaceMapFieldValue, true)
```

Comment:
we have to ouput the minimal desiredState to to do the change, if we don't filter we will output all the interfaces from the machine, and that can be massive and useless.

Reply:
If the user wants to filter, he can do a filtering operation before the replace.
But now I think about it, since `replaceMapFieldValue` always returns a value it doesn't matter if we pass here `true` or `false`. The result would be the same. But I find it confusing to pass `true`. We are not filtering anything.

Reply:
btw this is old code, we have to pass a pair of flags `shouldFilterSlice bool, shouldFilterMap bool`

Reply:
apart from filtering they will change the value with the output form the function we pass they do both things.

Reply:
The code is like this 
```golang
for _, valueToCheck := range originalSlice {                                
        value, err := applyFuncOnPath(valueToCheck, path, expectedNode, funcToApply, false, false)
        if err != nil {                                                         
            return nil, err                                                     
        }                                                                       
        if value != nil {                                                       
            filteredSlice = append(filteredSlice, value)                        
        }                                                                       
    }                                                                           
                                                                                
    if len(filteredSlice) == 0 {                                                
        return nil, nil                                                         
    }                                                                           
                                                                                
    if shouldFilterSlice {                                                      
        return filteredSlice, nil                                               
    }                              
```
If we don't pass shouldFilterSlice it does not use the value returned by the function.

It feels very ad-hoc, we can generalize at follow up.

Reply:
Got it, let's do only replace on replace, at the pipe PR we will tests it with the filter.